core/unit_test/TestNumericTraits.hpp

```diff
@@ -405,11 +405,10 @@ TEST(TEST_CATEGORY, numeric_traits_min_max_exponent10) {
   TestNumericTraits<TEST_EXECSPACE, long double, MaxExponent10>();
 #endif
 }
-TEST(TEST_CATEGORY, numeric_traits_quiet_and_signaling_nan) {
+
 // FIXME_NVHPC
-#ifdef KOKKOS_COMPILER_NVHPC
-  GTEST_SKIP() << "This test is known to fail with the NVHPC compiler";
-#endif
+#ifndef KOKKOS_COMPILER_NVHPC
+TEST(TEST_CATEGORY, numeric_traits_quiet_and_signaling_nan) {
```

Comment:
This is quite heavy handed, more like a last resort resolution.
What warning were we getting?

Reply:
I'll check each of the tests that are excluded.

Reply:
did you do this because you got warnings during the build from the test and the warnings were not fixable?

Reply:
To be clear, I didn't exclude this test as part of the PR, just changed how it was being excluded due to a warning. There are a few like this in the PR where you have 
```
TEST(TEST_CATEGORY, some_test) {
#ifdef KOKKOS_COMPILER_NVHPC  // FIXME: KOKKOS_COMPILER_NVHPC
  GTEST_SKIP() << "some error msg about not working with KOKKOS_COMPILER_NVHPC";
#endif

  // Test body
}
``` 
It complains that the `// Test body` code was not reachable because we are always exiting at `GTEST_SKIP()`. So instead, I'm ifdef-ing out the test (like is done other places in unit tests, I believe) and keeping the `// FIXME: KOKKOS_COMPILER_NVHPC

Reply:
So nothing is changed in this PR with what testing is run. I think addressing the test fails would make sense in a different PR.

Reply:
Any chance that you can use `pragma push/pop` or disable `unreachable` warnings generally in the CI?

Reply:
I'm sure we could do something like add `--diag_suppress=code_is_unreachable` flag, but would we want to ignore these instances in source code? Or do we not care if there is unreachable code?

Reply:
What Daniel is asking to see if we can selectively in those places disable the warning. Generally we prefer the gtest skip thing over ifdefing out the whole thing because it means the tests will show up as skipped instead of just not existing in gtest output.

Reply:
Ah ok, didn't realize the skips provided useful info. Yeah, let me find a nice solution.